### PR TITLE
Palette not optional

### PR DESCRIPTION
**Pull request by christiango**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Some recent changes to the styling package made palette and font optional, which means you'll need to null check if if you re using the strict options in Typescript. In reality, those types should always be defined. Updating the APIs that allow overriding the theme to take a Partial<ITheme> to make things work without making props optional. 
#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Theme/ThemePage.tsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ export class ThemePage extends React.Component<any, any> {
   private _onColorChanged(index: number, newColor: string) {
     let { colors } = this.state;
     let color = colors[index];
-    let theme: ITheme = {};
+    let theme: Partial<IPalette> = {};
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Theme/ThemePage.tsx]: I believe there's a bug here. Either this is a palette (in which case the theme variable should be renames) or we are in correctly creating an object to pass to loadTheme.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Theme/ThemePage.tsx]: Huh, you're right. Can you rename this variable to `pallete` or something? Then you can just `loadTheme({ pallete })

Also you're missing a change request for the styling package, so it wouldn't get published. maybe we need to add the check step.

Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Theme/ThemePage.tsx]: Yeah, I left it that way to make sure we noticed. I'll update it

**Comment by dzearing** [comment]
Awesome catch, sorry about that. Didn't even know about `Partial`!

**Pull request merged**